app/queries/monthly_revenue_query.rb

```diff
@@ -18,12 +18,13 @@ def initialize(account, options = { include_vat: true })
   end
 
   module CastValues
-    def cast_values(klass=Hash)
-      hash_rows.map { |row|
-        row.each_with_object(klass.new) { |(key, value), memo|
-          memo[key] = column_types.present? ? column_types[key].cast(value) : value
-        }
-      }
+    def cast_values(klass = Hash)
+      hash_rows.map do |row|
+        row.each_with_object(klass.new) do |(key, value), memo|
+          column_type = column_types[key]
+          memo[key] = column_type.present? ? column_type.cast(value) : value
```

Comment:
These are the lines that changed, the rest is just styling - multiline `{|args| }` vs `do |args| end`.

The description of why this is needed is in the commit description: https://github.com/3scale/porta/pull/3705/commits/128fa1c166078377f54d06fad606b75113f4ffe4

I didn't put a comment in the code, because I think it's not needed - it should work even when the original regression that broke this method is fixed in Rails 7.

Reply:
The question is whether because of this bug sometimes we may need to cast while the value will not exist. Didn't look very htoroughly to understand. But given how 61 worked for many people and that we have tests that caught it in the first place, lets hope it is solid.

Reply:
Hm, to be honest I didn't question the need to add this casting. I understand that the thing here is that some data types (e.g. Type::Integer, Type::Float, Type::Decimal, Type::String, Type::DateTime, Type::Boolean) are casted by the `pg` library itself, and others are not. But indeed, I'm not sure - maybe the latter ones are supposed to be casted automatically by Rails?... Could be :) but I didn't try that. Maybe for another PR.